ipam/calico-ipam.go

```diff
@@ -238,18 +244,25 @@ func cmdDel(args *skel.CmdArgs) error {
 		return fmt.Errorf("error constructing WorkloadEndpoint name: %s", err)
 	}
 
+	handleID, err := utils.GetHandleID(conf.Name, args.ContainerID, epIDs.WEPName)
+	if err != nil {
+		return err
+	}
+
 	logger := logrus.WithFields(logrus.Fields{
 		"Workload":    epIDs.WEPName,
 		"ContainerID": epIDs.ContainerID,
+		"HandleID":    handleID,
 	})
 
 	logger.Info("Releasing address using workloadID")
 	ctx := context.Background()
-	if err := calicoClient.IPAM().ReleaseByHandle(ctx, epIDs.WEPName); err != nil {
+	if err := calicoClient.IPAM().ReleaseByHandle(ctx, handleID); err != nil {
```

Comment:
I think we still need the old logic here in order to support upgrade from v2.6.4 to v3.0.